nucleus/dataset_item.py

```diff
@@ -0,0 +1,53 @@
+import os.path
+from .errors import FileNotFoundError
+from .constants import (
+    IMAGE_URL_KEY,
+    METADATA_KEY,
+    REFERENCE_ID_KEY,
+)
+
+
+class DatasetItem:
```

Comment:
@srikanth-scale FYI for docs, I added optional field for item_id here.  Best way to instantiate a DatasetItem now is:

`DatasetItem(s3://my-bucket/picture.jpg, reference_id = 'my_pic', metadata = {...})`